main/test/ca/uwaterloo/flix/language/phase/TestSafety.scala

```diff
@@ -125,4 +125,15 @@ class TestSafety extends FunSuite with TestUtils {
     expectError[IllegalNegativelyBoundWildcard](result)
   }
 
+  test("UseOfLatticeVariable.01") {
+    val input =
+      """
+        |pub def f(): #{ A(Int32), B(Int32; Int32) } = #{
+        |    A(x: Int32) :- B(12; x).
+        |}
+      """.stripMargin
+    val result = compile(input, Options.TestWithLibAll)
+    expectError[IllegalUseOfLatticeVariable](result)
+  }
```

Comment:
One test for now will do, we can add some more tests later (we have the ticket).